client/src/redux/selectors/todolistSelectors/index.ts

```diff
@@ -22,3 +30,85 @@ export const selectCompleted = createSelector(
     return list.filter((item: ToDoItemTypes) => item.completed);
   }
 );
+
+export const selectAllSortedList = createSelector(
+  [selectList, orderBy],
+  (list: TodoStateTypes['list'], order) => {
+    const newList = [...list];
+    return sortList(newList, order);
+  },
+  memoizeOptions
+);
+
+export const selectActiveSortedList = createSelector(
+  [selectActive, orderBy],
+  (list: TodoStateTypes['list'], order) => {
+    const newList = [...list];
+    return sortList(newList, order);
+  },
+  memoizeOptions
+);
+
+export const selectCompletedSortedList = createSelector(
+  [selectCompleted, orderBy],
+  (list: TodoStateTypes['list'], order) => {
+    const newList = [...list];
+    return sortList(newList, order);
+  },
+  memoizeOptions
+);
+
+function orderBy(state: RootState, order: string) {
+  return order;
+}
+
+function createdAsc(a: ToDoItemTypes, b: ToDoItemTypes) {
+  const first = new Date(a.created);
+  const second = new Date(b.created);
+  return first > second ? 1 : -1;
+}
+
+function createdDesc(a: ToDoItemTypes, b: ToDoItemTypes) {
+  const first = new Date(a.created);
+  const second = new Date(b.created);
+  return first < second ? 1 : -1;
+}
+
+function expiryAsc(a: ToDoItemTypes, b: ToDoItemTypes) {
+  const first = new Date(a.expireUntil);
+  const second = new Date(b.expireUntil);
+  return first > second ? 1 : -1;
+}
+
+function expiryDesc(a: ToDoItemTypes, b: ToDoItemTypes) {
+  const first = new Date(a.expireUntil);
+  const second = new Date(b.expireUntil);
+  return first < second ? 1 : -1;
+}
```

Comment:
I prefer to join it to one function
If you have 10 orders, will you create 10 functions ?

Reply:
Okey,thanks for your feedback.I will fix everything.

Reply:
I hope I understood you correctly, I will correct everything if necessary.